### PR TITLE
#166789882 Add user follow and unfollow resource

### PR DESCRIPTION
**Pull request by adafia**
#### What does this PR do?
 - add user follow another user
 - add user unfollow a user
 - add user view all their followers
 - add user view all users they follow

#### Description of Task to be completed?
 - A user should be able to follow another user and also unfollow them. 
 - A user should be able to view all the users who follow them
 - A user should be able to view all users they follow.

#### How should this be manually tested?
- These endpoints can be tested by:
1. Remotely:  accessing the swagger documentation via : `https://ah-lobos-backend-swagger.herokuapp.com`
2. Locally vial the following routes
> follow user: `/api/profiles/:username/follow` via `GET`
> unfollow user: `/api/profiles/:username/unfollow` via `DELETE`
> get all followers: `/api/profiles/followers` via `GET`
> get all users a user follows: `/api/profiles/followees` via `GET`

#### Any background context you want to provide?
- Users should be able to follow other users they are interested in and also unfollow others they loose interest in.
- Users should also have the option of seeing who they follow and also who follows them.

#### What are the relevant pivotal tracker stories?
[#166789882](https://www.pivotaltracker.com/story/show/166789882)
#### Screenshots (if appropriate)
- Swagger documentation 
 - Follow another user
Screen Shot 2019-07-18 at 21 00 16

Screen Shot 2019-07-18 at 21 00 54

 - Getting the users a user follows
Screen Shot 2019-07-18 at 21 01 50

Screen Shot 2019-07-18 at 21 02 13

 - Getting a user's followers
Screen Shot 2019-07-18 at 21 03 19

Screen Shot 2019-07-18 at 21 03 38

- unfollow a user
Screen Shot 2019-07-18 at 21 16 30

Screen Shot 2019-07-18 at 21 16 46

- Mockups
Screen Shot 2019-07-18 at 22 14 00

Screen Shot 2019-07-18 at 22 14 16

#### Questions:
N/A

### CODE REVIEW

**Comment by emukungu** [comment]
fix the eslint config to accommodate for these line breaks.

**Comment by adafia** [comment]
> fix the eslint config to accommodate for these line breaks.

okay i will.

**Review by manzif — APPROVED** [review]

**Review by Emile-Nsengimana — APPROVED** [review]

**Review by j4l13n — APPROVED** [review]

**Review by Dom58 — APPROVED** [review]

File: src/controllers/followersController.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+  /**
+   *
+   * @param {Object} req
+   * @param {Object} res
+   * @returns {Object} user successfully views all users that follow them
+   */
+  static async viewFollowers(req, res) {
+    const x = req.user.id;
+    const arr = await Followers.findAll({ where: { followee: x }, include: [{ model: Users, attributes: ['id', 'username', 'email', 'bio', 'image', 'favorites', 'following'] }] });
```
Comment (emukungu) [inline src/controllers/followersController.js]: make this line much more readable,
Layer out the code for all the files.

Check eslint for this. someone changed it to reject line breaks.

**Review by emukungu — CHANGES_REQUESTED** [review]

File: src/controllers/followersController.js (hunk 1 above)
Comment (adafia) [reply src/controllers/followersController.js]: Okay, I will.

**Comment by emukungu** [comment]
hound is complaining, fix it.

**Comment by adafia** [comment]
> hound is complaining, fix it.

@emukungu It's fixed now, please review it again

**Comment by emukungu** [comment]
instead of 'add' you can use 'allow' for your commit

**Comment by adafia** [comment]
> instead of 'add' you can use 'allow' for your commit

okay sure, i'll change it ASAP

**Comment by adafia** [comment]
> instead of 'add' you can use 'allow' for your commit

@emukungu it's done now

**Pull request merged**